src/blurb/blurb.py

```diff
@@ -946,6 +1047,23 @@ def init_tmp_with_template():
     print("Ready for commit.")
 
 
+assert sections, 'sections is empty'
+_sec_id_w = 2 + len(str(len(sections)))
+_sec_name_w = 2 + max(map(len, sections))
+_sec_rowrule = '+'.join(['',  '-' * _sec_id_w, '-' * _sec_name_w, ''])
+_format_row = ('| {:%d} | {:%d} |' % (_sec_id_w - 2, _sec_name_w - 2)).format
+sections_table = '\n'.join(itertools.chain(
+    [_sec_rowrule, _format_row('ID', 'Section'),_sec_rowrule.replace('-', '=')],
+    itertools.starmap(_format_row, enumerate(sections, 1)),
+    [_sec_rowrule]
+))
+del _format_row, _sec_rowrule, _sec_name_w, _sec_id_w
+add.__doc__ %= dict(
```

Comment:
I'm surprised you've submitted a PR using %-formatting in 2024.  If you remove the "section_example_id" (as I think section IDs are a bad idea), and hard-code the "section_example_name", the only remaining dynamically-computed text is the sections table, and that's appended, so you could just add that to the end of the string.

Reply:
Err... I think it's by habit when I write docstrings for argparse commands. I think argparse is still using the old %-style (or maybe not anymore). I'll update that and just use a .format() (I'd prefer not to use '+' just to be sure that newlines and spaces do not seem dangling and such that the docstring before being formatted makes sense).